tests/filecheck/dialects/x86/x86_assembly_section.mlir

```diff
@@ -0,0 +1,13 @@
+// RUN: XDSL_ROUNDTRIP
+
+x86.assembly_section ".text" {
+  x86.directive ".global" "external"
+  x86_func.func @external() -> ()
+}
+
+// CHECK:      builtin.module {
+// CHECK-NEXT:   x86.assembly_section ".text" {
+// CHECK-NEXT:     x86.directive ".global" "external"
+// CHECK-NEXT:     x86_func.func @external() -> ()
```

Comment:
Now that I'm thinking about it, can this be a simple `test.op`? I find it a bit confusing because this is an op that is a `DirectiveOp` and the assembly section also translates to a directive in plain assembly.